specifyweb/frontend/js_src/lib/components/DataModel/resourceApi.ts

```diff
@@ -747,13 +748,15 @@ export const ResourceBase = Backbone.Model.extend({
       return this;
     else if (this._fetch) return this._fetch;
     else
-      return (this._fetch = Backbone.Model.prototype.fetch
-        .call(this, options)
-        .then(() => {
-          this._fetch = null;
-          // BUG: consider doing this.needsSaved=false here
-          return this;
-        }));
+      return (this._fetch = hijackBackboneAjax(
+        options === undefined || options.strict ? undefined : [Http.NOT_FOUND],
+        () =>
+          Backbone.Model.prototype.fetch.call(this, options).then(() => {
+            this._fetch = null;
+            // BUG: consider doing this.needsSaved=false here
+            return this;
+          })
+      ));
```

Comment:
I'm not sure what the side-effects are if we make the 404 change in `backboneAjax.ts` I mentioned in the other comment. There doesn't seem to be any way to avoid the 404 redirect without using `hijackBackboneAjax()`. Without `hijackBackboneAjax()`, the error does not get passed up the chain high enough to reach QCBX. 

The error is initially caught here: https://github.com/specify/specify7/blob/6a364e00f96d5c81f0853c65b3d12ed0918fc202/specifyweb/frontend/js_src/lib/utils/ajax/response.ts#L76-L78

and in a prod env, it redirects to a 404 page and the error seems to remain uncaught:
https://github.com/specify/specify7/blob/013f81295c9e83cd6bd86a565d9b21950b5a76da/specifyweb/frontend/js_src/lib/components/Errors/FormatError.tsx#L152-L162

Reply:
> I'm not sure what the side-effects are if we make the 404 change in backboneAjax.ts I mentioned in the other comment. There doesn't seem to be any way to avoid the 404 redirect without using hijackBackboneAjax(). Without hijackBackboneAjax(), the error does not get passed up the chain high enough to reach QCBX.

I think you should be calling handleAjaxError() with expected errors [] not in this file but in the query combo box file. here: https://github.com/specify/specify7/blob/69391723254d4ea71df4701a869c225dc00b80dd/specifyweb/frontend/js_src/lib/components/QueryComboBox/index.tsx#L174-L175

that way, this change will not unexpectedly affect any other place, and you will have an error thrown like expected